src/state/sagas/datasets/FindAllDatasets/FindAllDatasets.js

```diff
@@ -16,37 +19,54 @@ const DATASET_PERMISSIONS_MAPPING = {
 };
 
 const getUserEmail = (state) => state.auth.userEmail;
+const keepOnlyReadableDatasets = (datasets) =>
+  datasets.filter(
+    (dataset) =>
+      dataset.security?.currentUserPermissions == null ||
+      dataset.security.currentUserPermissions?.includes(ACL_PERMISSIONS.DATASET.READ)
```

Comment:
Maybe '?' operator behind security as just above ?

Reply:
The first part of the condition handle the cases where security is `null` or `undefined`, we don't need the optional chaining operator in the second part :slightly_smiling_face: